OpenRA.Mods.Common/Projectiles/LaserZap.cs

```diff
@@ -32,13 +33,42 @@ public class LaserZapInfo : IProjectileInfo
 		[Desc("Equivalent to sequence ZOffset. Controls Z sorting.")]
 		public readonly int ZOffset = 0;
 
-		public readonly int BeamDuration = 10;
+		public readonly int Duration = 10;
 
 		public readonly bool UsePlayerColor = false;
 
-		[Desc("Laser color.")]
+		[Desc("Color of the beam.")]
 		public readonly Color Color = Color.Red;
 
+		[Desc("Beam follows the target.")]
+		public readonly bool TracksTarget = true;
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Extra search radius beyond beam width. Required to ensure affecting actors with large health radius.")]
+		public readonly WDist TargetExtraSearchRadius = new WDist(1536);
+
+		[Desc("Beam can be blocked.")]
+		public readonly bool Blockable = false;
+
+		[Desc("Draw a second beam (for 'glow' effect).")]
+		public readonly bool SecondaryBeam = false;
```

Comment:
In his defense, TS/RA2 did have two lasers drawn at once - LaserInnerColor and LaserOuterColor defined that. (TS/RA2 also had LaserOuterSpread though.)